### PR TITLE
Add Managed Identity support to Redis Cache

### PR DESCRIPTION
**Pull request by p3ck**
##### SUMMARY
Adds Managed Identity support to the rediscache module.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/azure_rm_rediscache.py
plugins/modules/azure_rm_rediscache_info.py
requirements.txt
tests/integration/targets/azure_rm_rediscache/tasks/main.yml

##### ADDITIONAL INFORMATION
Additional bug fixes to the test cases.  Removing invalid or deprecated options.

### CODE REVIEW

File: tests/integration/targets/azure_rm_rediscache/tasks/main.yml [hunk 1]
```diff
@@ -489,6 +529,22 @@
...
+    resource_group: "{{ resource_group }}"
+    provider: ManagedIdentity
+    resource_type: userAssignedIdentities
+    resource_name: "{{ item }}"
+    api_version: "2023-01-31"
+    state: absent
+  loop:
+    - "ansible-test-redis-identity"
+    - "ansible-test-redis-identity-2"
+
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_rediscache/tasks/main.yml]: Too many blank lines
```suggestion
```

File: plugins/modules/azure_rm_rediscache.py [hunk 2]
```diff
@@ -71,6 +71,42 @@
...
+                        required: false
+                        type: list
+                        elements: str
+                        default: []
+                    append:
+                        description:
+                            - If the list of identities has to be appended to current identities (true) or if it has to replace current identities (false)
+                        required: false
+                        type: bool
+                        default: False
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_rediscache.py]:
```suggestion
                        default: True
```

**Review by Fred-sun — COMMENTED** [review]
small change request!

File: plugins/modules/azure_rm_rediscache.py (hunk 2 above)
Comment (p3ck) [reply plugins/modules/azure_rm_rediscache.py]: So in #1643 you requested the default for append be False.  Here you are requesting we change the default to True.  

Can we agree on a default for all the modules?

Comment (Fred-sun) [reply plugins/modules/azure_rm_rediscache.py]: @p3ck As @xuzhang3  said, if default is false, we override the current identity by default. Thank you!

Comment (p3ck) [reply plugins/modules/azure_rm_rediscache.py]: If you agree with @xuzhang3 then the default here should be False.  If the default is False we will overwrite and only append if the user sets this to True

Comment (Fred-sun) [reply plugins/modules/azure_rm_rediscache.py]: @p3ck Ok, I agree!

Comment (Fred-sun) [inline plugins/modules/azure_rm_rediscache.py]: @p3ck  doc-default-does-not-match-spec: Argument 'append' in argument_spec found in identity -> user_assigned_identities defines default as (True) but documentation defines default as (False)

**Review by Fred-sun — COMMENTED** [review]
small change request!

File: plugins/modules/azure_rm_rediscache.py (hunk 2 above)
Comment (Fred-sun) [inline plugins/modules/azure_rm_rediscache.py]: @p3ck The argument default value is ['true',](url https://github.com/ansible-collections/azure/blob/997e4de875fa12e99c9a064b614797a65c58677e/plugins/module_utils/azure_rm_common_ext.py#L37) Please keep consistent. Thank you!

**Review by Fred-sun — COMMENTED** [review]
Small change requst!

File: plugins/modules/azure_rm_rediscache.py (hunk 2 above)
Comment (p3ck) [reply plugins/modules/azure_rm_rediscache.py]: done

File: tests/integration/targets/azure_rm_rediscache/tasks/main.yml (hunk 1 above)
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_rediscache/tasks/main.yml]: Too many blank lines, Please delete the blank lines!
```suggestion
```

Comment (p3ck) [reply tests/integration/targets/azure_rm_rediscache/tasks/main.yml]: Thanks, Done.

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**